pkg/apis/networking/types.go

```diff
@@ -53,8 +53,10 @@ type NetworkPolicySpec struct {
 	// The array of ingress rules is applied to any pods selected by this field.
 	// Multiple network policies can select the same set of pods. In this case,
 	// the ingress rules for each are combined additively.
-	// This field is NOT optional and follows standard label selector semantics.
-	// An empty podSelector matches all pods in this namespace.
+	// This field is optional. If it is not specified, it defaults to an empty selector,
+	// which matches all pods in this namespace.
+	// An explicitly specified empty selector also matches all pods.
+	// +optional
```

Comment:
Hm. I don't like talking about the default and then going back to add more detail about the non-default case.

Also, the existing text turns out to have _another_ bug; it shouldn't say "ingress rules" since this applies to egress as well.

So how about

```
	// podSelector selects the pods to which this NetworkPolicy object applies.
	// The array of rules is applied to any pods selected by this field. An empty
	// selector matches all pods in the policy's namespace.
	// Multiple network policies can select the same set of pods. In this case,
	// the ingress rules for each are combined additively.
	// This field is optional. If it is not specified, it defaults to an empty selector.
	// +optional
```

and then the same in the other copy.